sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowCreateTableExec.scala

```diff
@@ -57,7 +60,7 @@ case class ShowCreateTableExec(
   }
 
   private def showTableDataColumns(table: Table, builder: StringBuilder): Unit = {
-    val columns = table.schema().fields.map(_.toDDL)
+    val columns = CharVarcharUtils.getRawSchema(table.schema(), SQLConf.get).fields.map(_.toDDL)
```

Comment:
```suggestion
    val columns = CharVarcharUtils.getRawSchema(table.schema(), conf).fields.map(_.toDDL)
```
We are within a physical plan, we can access `conf` here.